rhel/vex/parser.go

```diff
@@ -290,8 +272,46 @@ func (c *creator) knownAffectedVulnerabilities(ctx context.Context, v csaf.Vulne
 			return nil, fmt.Errorf("could not unbind cpe: %s %w", ch, err)
 		}
 		vuln.Repo = c.rc.Get(wfn)
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			if err != nil {
+				zlog.Warn(ctx).
+					Str("purlHelper", purlHelper).
+					Err(err).
+					Msg("could not parse PURL")
+			}
+			if !checkPURL(purl) {
+				continue
+			}
+			pkgName, err = extractPackageName(purl)
```

Comment:
`extractPackageName` returns `""` for `pkgName` when `err != nil`, and I think the hope is to leave it as-is